lib/phoenix/socket.ex

```diff
@@ -47,6 +117,135 @@ defmodule Phoenix.Socket do
             topic: nil,
             transport: nil,
             transport_pid: nil
+
+
+  defmacro __using__(_) do
+    quote do
+      @behaviour Phoenix.Socket
+      import unquote(__MODULE__)
+      Module.register_attribute(__MODULE__, :phoenix_channels, accumulate: true)
+      @phoenix_transports %{}
+
+      transport :websocket, Phoenix.Transports.WebSocket,
+        serializer: Phoenix.Transports.JSONSerializer,
+        timeout: :infinity
+
+      transport :longpoll, Phoenix.Transports.LongPoll,
```

Comment:
I just think it should be commented to push people to not rely on it if they don't need to. Then it is easier for us to move it elsewhere in the future.